dnfdaemon-server/services/goal/goal.cpp

```diff
@@ -52,7 +52,7 @@ void packages_to_transaction(
     const libdnf::transaction::TransactionItemAction action,
     const std::vector<libdnf::rpm::Package> & packages) {
     std::vector<std::string> attr{
-        "name", "epoch", "version", "release", "arch", "repo", "package_size", "install_size", "full_nevra"};
+        "name", "epoch", "version", "release", "arch", "repo", "package_size", "install_size", "evr"};
```

Comment:
Also, the attribute list could be simplified in the future removing other attr that are not explicitly called in the code such as `version` and `release`. In the end the attr list could become something like:
```
"name", "arch", "repo", "package_size", "install_size", "evr", "full_nevra"
```
Which as @m-blaha suggested it is the minimal info reuqired for this step.
Another solution is leaving `version`, `epoch`, etc. in the attr list and compute the `full_nevra` when needed.
What do you think?

Reply:
I've rewritten the sort function to a cmp function and passed that to std::sort().
The new cmp function compares by name, arch and evr (using rpm's rpmvercmp).
It's even better than comparing by NEVRAs, because in this case the data is ordered exactly according to the displayed fields.

Reply:
thank you. I will test it and then looks good to me!